File: MobileMeteo/MMTMeteorogramStore.m
//
//  MMTMeteorogramStore.m
//  MobileMeteo
//
//  Created by Kamil on 07.06.2015.
//  Copyright (c) 2015 Kamil Szostakowski. All rights reserved.
//

#import "MMTMeteorogramStore.h"

@implementation MMTMeteorogramStore

- (void)getMeteorogramWithQuery:(MMTMeteorogramQuery*)query completion:(MMTMeteorogramFetchCompletion)completion
{
    MMTMeteorogramFetchDelegate *delegate = [[MMTMeteorogramFetchDelegate alloc] initWithCompletion:completion];
    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL mmt_meteorogramSearchUrlWithQuery:query]];
    
    [[[NSURLConnection alloc] initWithRequest:request delegate:delegate] start];
}

@end

File: MobileMeteo/MMTMeteorogramStore.swift
//
//  MMTUmMeteorogramStore.swift
//  MobileMeteo
//
//  Created by Kamil Szostakowski on 19.08.2015.
//  Copyright (c) 2015 Kamil Szostakowski. All rights reserved.
//

import Foundation
import CoreLocation

class MMTMeteorogramStore: MMTForecastStore
{    
    // MARK: Methods
    
    func getMeteorogramForLocation(_ location: CLLocation, completion: @escaping MMTFetchMeteorogramCompletion)
    {
        urlSession.fetchMeteorogramImageForUrl(try! URL.mmt_meteorogramSearchUrl(for: climateModel.type, location: location), completion: completion)
    }
    
    func getMeteorogramLegend(_ completion: @escaping MMTFetchMeteorogramCompletion)
    {        
        urlSession.fetchImageFromUrl(try! URL.mmt_meteorogramLegendUrl(for: climateModel.type), completion: completion)
    }
}
